refactor: Made the colors match the Windows theme

meatsnails opened the pull request:
Changed the way the ui color's are handled to be able to be chosen based off the windows system theme (light/dark mode) at runtime this both effects the upload history and the upload status menu and below is an example of both light and dark mode
<img width="422" height="192" alt="image" src="https://github.com/user-attachments/assets/49be9118-64fb-42ff-9445-e44bff697562" />
<img width="422" height="192" alt="image" src="https://github.com/user-attachments/assets/294452f6-961d-4aa7-8d1d-31cbe2a83485" />

meatsnails committed:
refactor: Made the colors match the Windows theme

meatharvester commented:
the icons could be changed though that isn't in my ability

nesuwu reviewed (approved):
looked at the logic and moving colors to dictionaries is a huge win for readability much cleaner than before, the registry check logic is solid for win10/11 just note that the theme only sets on startup so it wont auto switch if you change settings while running
also in create_thumbnail line 490 that bare except catches literally everything including keyboard interrupts so sticking to except exception is safer otherwise solid refactor

karimawi commented:
No worries guys, just got super busy working on my graduation project, will review and merge in a few hours, thank you 🙏

Commit:
Improves upload cancellation and UI theming

Adds functionality to properly cancel uploads and provides theme-aware icons.

This commit introduces an `UploadCancelledException` to handle upload cancellations gracefully. It ensures that uploads can be terminated cleanly by closing the requests session and updating the UI accordingly.  UI updates are now handled after cancellation.

Additionally, it implements theme-aware icons for a better user experience, switching between light and dark variants based on the Windows theme.  Also adds new theme styling for menus.

Removes `terminate()` usage on the upload worker thread, since it does not guarantee the thread stops. Instead, introduces a flag and session management to cancel uploads properly.
